src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,80 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import React, { Dispatch, SetStateAction } from 'react';
+
+type Props = {
+  currentPage: number;
+  total: number;
+  postPerPage: number;
+  onPageChange:(pageNumber: number) => void;
+  setCurrentPage:Dispatch<SetStateAction<number>>;
+
+};
+
+export const Pagination: React.FC<Props>
+= ({
+  currentPage, total, postPerPage, onPageChange, setCurrentPage,
+}) => {
+  const pageNumbers = [];
+
+  for (let i = 1; i <= Math.ceil(total / postPerPage); i += 1) {
+    pageNumbers.push(i);
+  }
+
+  const previousPage = () => {
+    setCurrentPage((current: number) => current - 1);
+  };
+
+  const nextPage = () => {
+    setCurrentPage((current: number) => current + 1);
+  };
+
+  return (
+    <ul className="pagination">
+      <li className={classNames(
+        'page-item',
+        { disabled: currentPage === 1 },
+      )}
+      >
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          onClick={previousPage}
+          aria-disabled="true"
```

Comment:
```suggestion
          aria-disabled={currentPage === 1}
```

but better move it to a separate variable. it will fix a few tests (the same for `nextLink`)